src/System.Text.JsonLab/System/Text/Json/JsonWriter.cs

```diff
@@ -568,22 +1511,237 @@ private void WriteSpacingUtf16(bool newline)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private Span<byte> EnsureBuffer(int needed = 256)
```

Comment:
Yeah, as I said above, I am not thrilled about all the calls to Ensure, Calculate, and other checks. I think we should write everything with Try methods, as in most cases the buffer will be large enough.